### ISSUE TITLE
[unified-signatures] false positive on async functions

### ISSUE THREAD

**Issue by mysticatea** [typescript-eslint/typescript-eslint#452]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/unified-signatures": "error"
  }
}
```

```TS
interface Table<T> {
    getById(ids: ReadonlyArray<string>): Promise<Map<string, T>>
    getById(id: string): Promise<T | undefined>
}
```

**Expected Result**

No errors.

**Actual Result**

```
3:13  error  These overloads can be combined into one signature taking `ReadonlyArray<string> | string`  @typescript-eslint/unified-signatures
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.7.0` |
| `@typescript-eslint/parser`        | `1.7.0` |
| `TypeScript`                       | `3.4.3` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `10.15.3` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
This was fixed in #839
